jsengine-ios/jsengine/InterceptorURLProtocol.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  InterceptorURLProtocol.swift
+//  jsengine
+//
+//  Created by Ghadir Eraisha on 12/14/16.
+//  Copyright © 2016 Cliqz GmbH. All rights reserved.
+//
+
+import UIKit
+
+class InterceptorURLProtocol: NSURLProtocol {
+    
+    static let customURLProtocolHandledKey = "customURLProtocolHandledKey"
+    static let excludeUrlPrefixes = ["https://lookback.io/api", "http://localhost"]
+    
+    //MARK: - NSURLProtocol handling
+    override class func canInitWithRequest(request: NSURLRequest) -> Bool {
+        guard (
+            NSURLProtocol.propertyForKey(customURLProtocolHandledKey, inRequest: request) == nil
+                && request.mainDocumentURL != nil) else {
+                    return false
+        }
+        guard isExcludedUrl(request.URL) == false else {
+            return false
+        }
+        //TODO: exclude HTTPHandler Request from being intercepted
+        if let webRequest = Engine.sharedInstance.webRequest {
+            return webRequest.shouldBlockRequest(request)
+        }
+        
+        return false
+    }
+    
+    override class func canonicalRequestForRequest(request: NSURLRequest) -> NSURLRequest {
```

Comment:
Why do we need to override this method? How does this implementation differ from the super class one?